routes/cube_routes.js

```diff
@@ -1186,199 +1114,67 @@ router.post('/uploaddecklist/:id', ensureAuth, async (req, res) => {
+          for (let j = 0; j < count; j++) {
+            cards.push(numericMatch[2]);
+          }
         } else {
-          missing += `${item}\n`;
+          let selected = null;
+          if (/(.*)( \((.*)\))/.test(item)) {
+            // has set info
+            const name = item.substring(0, item.indexOf('('));
+            const potentialIds = carddb.getIdsFromName(name);
+            if (potentialIds && potentialIds.length > 0) {
+              const set = item.toLowerCase().substring(item.indexOf('(') + 1, item.indexOf(')'));
+              // if we've found a match, and it DOES need to be parsed with cubecobra syntax
+              const matching = potentialIds.find((id) => carddb.cardFromId(id).set.toUpperCase() === set);
+              selected = matching || potentialIds[0];
+            }
+          } else {
+            // does not have set info
+            const selectedCard = carddb.getMostReasonable(item, cube.defaultPrinting);
+            selected = selectedCard ? selectedCard._id : null;
+          }
+          if (selected) {
+            const details = carddb.cardFromId(selected);
+            if (!details.error) {
+              util.addCardToCube(cube, details);
+              added.push(details);
+              changelog += addCardHtml(details);
+            } else {
+              missing += `${item}\n`;
+            }
+          }
         }
-      } else {
-        missing += `${item}\n`;
       }
     }
   }
-
-  const blogpost = new Blog();
-  blogpost.title = 'Cube Bulk Import - Automatic Post';
-  blogpost.html = changelog;
-  blogpost.owner = cube.owner;
-  blogpost.date = Date.now();
-  blogpost.cube = cube._id;
-  blogpost.dev = 'false';
-  blogpost.date_formatted = blogpost.date.toLocaleString('en-US');
-  blogpost.username = cube.owner_name;
-  blogpost.cubename = cube.name;
-
-  if (missing.length > 0) {
-    const reactProps = {
-      cubeID: req.params.id,
-      missing,
-      added: added.map(({ _id, name, image_normal, image_flip }) => ({ _id, name, image_normal, image_flip })),
-      blogpost: blogpost.toObject(),
-    };
-    return res.render('cube/bulk_upload', {
-      reactHTML: BulkUploadPage
-        ? await ReactDOMServer.renderToString(React.createElement(BulkUploadPage, reactProps))
-        : undefined,
-      reactProps: serialize(reactProps),
-      cube,
-      cubeID: req.params.id,
-      title: `${abbreviate(cube.name)} - Bulk Upload`,
-    });
-  }
-
-  try {
-    await blogpost.save();
-    cube = setCubeType(cube, carddb);
-    await Cube.updateOne(
-      {
-        _id: cube._id,
-      },
-      cube,
-    );
-    req.flash('success', 'All cards successfully added.');
-    return res.redirect(`/cube/list/${req.params.id}`);
-  } catch (err) {
-    req.logger.error(err);
-    req.flash('danger', 'Error adding cards. Please try again.');
-    return res.redirect(`/cube/list/${req.params.id}`);
-  }
+  return updateCubeAndBlog(req, res, cube, changelog, added, missing);
```

Comment:
needs `await`

Reply:
This is an async function so you don't need to await the return value, it's an eslint error to put an await here.

Reply:
If it's the last line in the route, use await instead of return